src/blog/2023/09/modernize-your-legacy-industrial-data-part2.md

```diff
@@ -0,0 +1,137 @@
+description: A deeper dive with into making the most of legacy industrial data from the likes of Modbus and older, non IIoT protocols and putting it to work in an IIoT world.
+date: 2023-09-17
+authors: ["stephen-mclaughlin"]
+image: "/blog/2023/09/images/industrial-legacy-data-blog-image.png"
+tags:
+    - posts
+    - node-red
+    - community
+---
+
+In an article published recently, I introduced the topic of working with legacy industrial data from the likes of Modbus and older, non IIoT protocols and putting it to work in an IIoT world. 
+We looked at some of the challenges and how Node-RED with `node-red-contrib-buffer-parser` node can help. 
+
+In this article, I will dive a little deeper into the topic and discuss some of the finer details. I hope to demonstrate a smarter approach that can make a huge difference to data accuracy, performance and maintainability while significantly reducing developer time. Not only that, ending with a no-code solution.
+
+See the [previous article](modernize-your-legacy-industrial-data-part2.md) for the introduction and overview.
+
+<!--more-->
+
+### In the beginning...
+
+In order to generate IIoT data from legacy systems we first need to _grab_ that data.
+
+Node-RED has core nodes that can help you and many more contribution nodes exist that provide access to a wide range of industrial devices. To give you an idea, `node-red-contrib-modbus`, `node-red-contrib-s7comm`, `node-red-contrib-omron-fins`, `node-red-contrib-mcprotocol`, `node-red-contrib-df1` and `node-red-contrib-cip-st-ethernet-ip` are just some of the PLC data access nodes available.
+
+But getting the data is just the beginning, it's the methods and considerations you need to make that can make the difference between success and failure. Read on...
+
+
+### Data consistency
+
+An often overlooked aspect of working with legacy industrial data is the consistency of the data being read. In the context of this article, consistency means that the multiple values that make up a related data set are read in a way that they are all valid to one another at the point in time it was read.
+
+Let's take a look at a simple example. We have a process PLC recording production metrics and wish to get this data from its Modbus interface for reporting and decision making. The PLC has 5 values that we need to read:
+
+| Register | Value | Description            | Data Type  |
+| -------- | ----- | ---------------------- | ---------- |
+| 1, 2     | 10    | Part Count             | UINT32     |
+| 3, 4     | 2.5   | Cycle Time (sec)       | UINT32/100 |
+| 5, 6     | 30    | Production Time (sec)  | UINT32/100 |
+| 7, 8     | 20    | Run Time (sec)         | UINT32/100 |
+| 9, 10    | 11    | Stoppage Time (sec)    | UINT32/100 |
+
+If the total production time is 30 seconds and the run time is 20 seconds, then the stoppage time should be 10 seconds. But it is not. It is 11 seconds. This is because the data is not consistent.
```

Comment:
```suggestion
If the total production time is 30 seconds and the run time is 20 seconds, then the stoppage time should be 10 seconds. But it might not be, it could be 11 seconds. This is because the data is not consistent.
```

Reply:
Not sure about this suggestion. I was describing the actual data in the table above, so in that instance it actually was 11.  I understand why you raised this so will attempt to clarify.

Reply:
Applied change to clarify relationship between the sentence and sample data in [38834ba](https://github.com/flowforge/website/pull/1123/commits/38834bab3664f688aade56cdaebbe0c7175a7d9d)